testing/python/collect.py

```diff
@@ -274,7 +274,7 @@ def test_function_as_object_instance_ignored(self, pytester: Pytester) -> None:
         pytester.makepyfile(
             """
             class A(object):
-                def __call__(self, tmpdir):
+                def __call__(self):
```

Comment:
This intentionally has a fixture name for the signature

Reply:
> This intentionally has a fixture name for the signature

Ah, in that case would you like it renamed `tmpdir->tmp_path` (I've been proceeding under the assumption that even for non-functional use-cases there's value in renaming just to make things clearer).

Out of curiosity, what's the intent in having a fixture name in the signature? I couldn't see anything explicitly wanting it in the test (though admittedly I didn't dive too deep into the surrounding context)

Reply:
+1 for renaming it to `tmp_path` for consistency.

As for your question - since the test is named `test_function_as_object_instance_ignored` (and also looking at 24db492f53a7391c2bbc660adb0c44f072dfca58), the test seems to ensure that objects named `test_x` aren't collected. Part of that collection is also filling fixture values, so we'd want to make sure that this doesn't break.

Reply:
> As for your question - since the test is named test_function_as_object_instance_ignored (and also looking at 24db492), the test seems to ensure that objects named test_x aren't collected. Part of that collection is also filling fixture values, so we'd want to make sure that this doesn't break.

Thanks, do you think there would be any value in being more explicit about this? e.g.

```python
            import pytest

            @pytest.fixture
            def never_filled():
                assert False, "Fixture should never be filled!"

            class A(object):
                def __call__(self, never_filled):
                    0/0

            test_a = A()
```

Reply:
:+1: restored + renamed c9bb4c4

Reply:
I don't really have an opinion on making it more explicit vs. the way it is currently. Surely wouldn't hurt, but I don't mind much either way.